public/app/types/explore.ts

```diff
@@ -182,6 +182,11 @@ export interface ExploreItemState {
 
   queryResponse: PanelData;
 
+  /**
+   * Stores previous time ranges from ShowMoreLogs
+   */
+  prevShowMoreLogsTimeRanges: TimeRange[];
```

Comment:
This needs some explanation it is not clear if `prev` means just from the stand point of logs timeline or previous state of that.

Reply:
It's `prev` as for the previous state time ranges. Gonna make the description more precise.

Reply:
But after we run the append query this actually contains the current range doesn't it? Like last element here is the current range shown on the screen? Feels like something more vague like timeRangesStack would make more sense with some better comments.

Reply:
Yep, last element is the current range shown on the screen. Good idea in terms of renaming it to `timeRangesStack` - changed.